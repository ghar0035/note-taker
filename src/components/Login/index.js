import axios from "axios";
import { useState } from "react";
import { useNavigate } from "react-router-dom";
import Header from "../common/header";
import { Container } from "../Register";

const Login = () => {
  const [user, setUser] = useState({
    email: "",
    password: "",
  });
  const [error, setError] = useState("");
  const navigate = useNavigate()

  const onHandleChange = (e) => {
    const { name, value } = e.target;
    setUser((prevUser) => ({
      ...prevUser,
      [name]: value,
    }));
  };

  const handleSubmit = (event) => {
    event.preventDefault();

    axios
      .post("http://127.0.0.1:8001/login", user)
      .then((res) => {
        window.localStorage.setItem('user', JSON.stringify({
          userId: res.data.id,
          email: user.email,
          name: res.data.name
        }));

        setTimeout(() => {
          navigate('/dashboard')
        });
        
      })
      .catch((err) => {
        if (err.request.status === 401) {
          setError("Username or password is incorrect");
        }
      });
  };
  
  return (
    <>
      <Header />
      <div className="container">
        <Container>
          <form onSubmit={handleSubmit}>
            <div className="form-control">
              <label htmlFor="name">Email</label>
              <input
                id="email"
                name="email"
                type="email"
                onChange={onHandleChange}
                value={user.email}
              />
            </div>

            <div className="form-control">
              <label htmlFor="password">Password</label>
              <input
                id="password"
                name="password"
                type="password"
                onChange={onHandleChange}
                value={user.password}
              />
            </div>

            <button type="submit" className="btn">
              Login
            </button>

            <p className="error">{error}</p>
          </form>
        </Container>
      </div>
    </>
  );
};

export default Login;
