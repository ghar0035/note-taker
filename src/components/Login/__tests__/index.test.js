import renderer from "react-test-renderer";
import Login from "../index.js";
import { MemoryRouter as Router } from "react-router-dom";

test("renders Login", () => {
    const tree = renderer
      .create(
        <Router>
          <Login />
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });