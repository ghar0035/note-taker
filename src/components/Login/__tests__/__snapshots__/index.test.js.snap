// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Login 1`] = `
Array [
  <div
    className="sc-bcXHqe evUqyv"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <img
        alt="logo"
        className="logo"
        src="logo.png"
      />
    </a>
    <div>
      <a
        className="lnkBtn"
        href="/auth/register"
        onClick={[Function]}
      >
        register
      </a>
      <a
        className="lnkBtn primary"
        href="/auth"
        onClick={[Function]}
      >
        login
      </a>
    </div>
  </div>,
  <div
    className="container"
  >
    <div
      className="sc-gswNZR kNyIcL"
    >
      <form
        onSubmit={[Function]}
      >
        <div
          className="form-control"
        >
          <label
            htmlFor="name"
          >
            Email
          </label>
          <input
            id="email"
            name="email"
            onChange={[Function]}
            type="email"
            value=""
          />
        </div>
        <div
          className="form-control"
        >
          <label
            htmlFor="password"
          >
            Password
          </label>
          <input
            id="password"
            name="password"
            onChange={[Function]}
            type="password"
            value=""
          />
        </div>
        <button
          className="btn"
          type="submit"
        >
          Login
        </button>
        <p
          className="error"
        />
      </form>
    </div>
  </div>,
]
`;
